make/tools.mk

```diff
@@ -56,7 +56,7 @@ KUBEBUILDER_ASSETS_VERSION=1.27.1
 TOOLS += etcd=$(KUBEBUILDER_ASSETS_VERSION)
 TOOLS += kube-apiserver=$(KUBEBUILDER_ASSETS_VERSION)
 
-VENDORED_GO_VERSION := 1.23.7
+VENDORED_GO_VERSION := 1.23.8
```

Comment:
There are no Go checksums in the release-1.12, unlike release-1.16 and release-1.17. E.g.
 * https://github.com/cert-manager/cert-manager/pull/7706/commits/a0b43b2c3b699438c0a3d0bbc27b021ee665b421

I guess the checksums were introduced when we adopted the makefile-modules